### ISSUE TITLE
Add more config options in extension's setting

### ISSUE THREAD

**Issue by ra-jeev** [ra-jeev/write-assist-ai#3]
Right now the model name, as well as some API request params (temperature / max_p etc) are hardcoded in the extension. These can be exposed in extension's settings so that the user can play around with different combinations.

**Comment by vsukhoml** [comment]
Instruction prompt can also be one of those params, say to provide more context about document and additional instructions.  It can be inserted before the ``${prompt}. ${subPrompt}:\n\n${text}`,

**Comment by vsukhoml** [comment]
I noticed only gpt-3.5-instruct models - does it mean that models like `gpt-4-turbo-preview` don't work well?

**Comment by ra-jeev** [comment]
> I noticed only gpt-3.5-instruct models - does it mean that models like `gpt-4-turbo-preview` don't work well?

@vsukhoml Right now the extension is using the legacy `v1/completions` endpoint of `OpenAI`. For these endpoints only the `instruct` models work (along with the older `davinci-002` and `babbage-002` models which didn't seem to follow the instructions so I haven't included). More [details here](https://platform.openai.com/docs/guides/text-generation/completions-api).

I do plan to move to the newer `v1/chat/completions` endpoints soon, and then the other models will work.

**Comment by vsukhoml** [comment]
Got it, thanks a lot for clarification!

**Comment by ra-jeev** [comment]
> Got it, thanks a lot for clarification!

No problem.  

Were you able to try out the new release? If you face any issue, please do let me know. 
Apart from the config options, one other major point in the release was: securing the API Key (removed it from the `settings.json`&mdash;so it won't be visible in VSCode settings now&mdash;and moved it to VSCode's `secretStorage`).

**Comment by vsukhoml** [comment]
Yes, it updated yesterday and seems to work fine. When I increased temperature I was able to get more varying outputs.

Moving key to `secretStorage` is a very good solution.
